sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from "@nestjs/common";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextValidationResult,
+} from "./application-bulk-withdrawal-import-text.models";
+import { DataSource, In } from "typeorm";
+import { Application, RecordDataModelService } from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalImportBusinessValidationModel,
+  ApplicationData,
+} from "./application-bulk-withdrawal-import-business-validation.models";
+
+type ApplicationSINDataMap = Record<string, ApplicationData>;
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportBusinessService extends RecordDataModelService<Application> {
+  constructor(private readonly dataSource: DataSource) {
+    super(dataSource.getRepository(Application));
+  }
+
+  /**
+   * Generates the application bulk withdrawal business validation model.
+   * @param textValidations text validation models to be converted to the business validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @param allowedLocationIds set of allowed location ids for the given institution.
+   * @returns application bulk withdrawal business models to be validated and persisted.
+   */
+  async generateBusinessValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+    allowedLocationIds: number[],
+  ): Promise<ApplicationBulkWithdrawalImportBusinessValidationModel[]> {
+    const applicationSINDataMap: ApplicationSINDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationSINData = await this.getSINForApplications(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationSINData.forEach((record) => {
+      applicationSINDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[] =
+      [];
+    textValidations.forEach((textValidation) => {
```

Comment:
Not a blocker but it seems that it can be done as a `map`.

Reply:
@andrewsignori-aot Is it better to use `map` compared to `forEach` here? I felt `forEach` to be a better fit here since we are populating `businessValidationModels` array rather than updating `textValidations` array. Please let me know if I missed something.

Reply:
`map` is intended to be used exactly to generate an array based on a second array, transforming the last one, which means, creating the `businessValidationModels` based on the `textValidations` where there will an one-to-one output.
I would not state this as a blocker but I have a feeling that along the solution whatever a `map` can be used we have been opting for it instead of a `foreach`.
Reason that would lead me to use the `foreach`:
- Performance (not the case IMO)
- Code readability (not the case IMO)

Reply:
Changed it.